test/fuzzers/frame_fuzzer.cpp

```diff
@@ -0,0 +1,119 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+#include <folly/String.h>
+#include <folly/io/async/EventBaseManager.h>
+#include <iostream>
+
+#include <folly/init/Init.h>
+#include <folly/portability/GFlags.h>
+#include <glog/logging.h>
+
+#include "rsocket/ConnectionAcceptor.h"
+#include "rsocket/DuplexConnection.h"
+#include "rsocket/RSocketServer.h"
+
+struct FuzzerConnectionAcceptor : public rsocket::ConnectionAcceptor {
+  void start(rsocket::OnDuplexConnectionAccept func) override {
+    VLOG(1) << "FuzzerConnectionAcceptor::start()" << std::endl;
+    func_ = func;
+  }
+
+  void stop() override {
+    VLOG(1) << "FuzzerConnectionAcceptor::stop()" << std::endl;
+  }
+
+  folly::Optional<uint16_t> listeningPort() const override {
+    return {0};
+  }
+
+  rsocket::OnDuplexConnectionAccept func_;
+};
+
+struct FuzzerDuplexConnection : public rsocket::DuplexConnection {
+  using Subscriber = rsocket::DuplexConnection::Subscriber;
+
+  struct SinkSubscriber : public Subscriber {
+    std::vector<std::unique_ptr<folly::IOBuf>> sunk_buffers_;
+
+    void onNext(std::unique_ptr<folly::IOBuf> buf) {
+      VLOG(1) << "SinkSubscriber::onNext(\""
+              << folly::humanify(buf->cloneAsValue().moveToFbString()) << "\")"
+              << std::endl;
+      sunk_buffers_.push_back(std::move(buf));
+    }
+  };
+
+  FuzzerDuplexConnection() : output_sub_(yarpl::make_ref<SinkSubscriber>()) {}
+
+  virtual void setInput(yarpl::Reference<Subscriber> sub) {
+    VLOG(1) << "FuzzerDuplexConnection::setInput()" << std::endl;
+    input_sub_ = sub;
+  }
+  virtual yarpl::Reference<Subscriber> getOutput() {
+    VLOG(1) << "FuzzerDuplexConnection::getOutput()" << std::endl;
+    return output_sub_;
+  }
+
+  yarpl::Reference<Subscriber> input_sub_;
```

Comment:
Nit: Ditto about trailing `_`.  I also think these should be camelCase to match the rest of the codebase.